### DISCUSSION TITLE
Rank fusion using Pyserini

### DISCUSSION THREAD

**Discussion by Fatima-Haouari** [castorini/pyserini#924] [Q&A] [answered]
Kindly direct me to examples on how to use Pyserini to create fused runs using reciprocal_rank_fusion and taking the average/sum of scores.

**Comment by lintool** [comment, marked as answer]
We don't have the documentation written up, but consult this test case for usage: https://github.com/castorini/pyserini/blob/master/integrations/sparse/test_verify_rrf.py

A contribution to the documentation would be welcome.

**Reply by Fatima-Haouari** [reply]
Thanks a lot for you help. I managed to create my fused runs now. However I have the following issue. 
The returned fused runs have the document IDs in float format (not string) which is causing a formatting issue with long documents IDs such the example below:
938526354907201539 Q0 **1.2399882500827095e+18** 10 11.35319995880127 interpolation
Kindly advice

**Reply by lintool** [reply]
Yea, there's some bug in the code resulting from Python duck typing. Please file an issue with a minimal example. It'd be even better if you could submit a patch that would fix the bug, complete with a test case. My guess is that the fix would be pretty simple, just an explicit cast somewhere...
